falcon/errors.py

```diff
@@ -272,6 +272,30 @@ def __init__(self, title, description, retry_after=None, **kwargs):
                                                         **kwargs)
 
 
+class HTTPUriTooLong(HTTPError):
```

Comment:
I was going to suggest capitalizing "URI" per pep8, i.e.: `HTTPURITooLong`. 

> Note: When using abbreviations in CapWords, capitalize all the letters of the abbreviation. Thus HTTPServerError is better than HttpServerError.

However, all the caps in `HTTPURI` tend to run together, so `Uri` may actually be better in this case.